### PR TITLE
Condense 6  methods to reduce total lines by over 40

### PR DESCRIPTION
**Pull request by denverdevelopments**
PUSH#2:  2/28__Delegate to new 'game' class methods (to act like helpers), and more refactoring several methods.

PUSH#1:  2/27__Reworked/condensed the following to save over 40 lines, but keep function:  
  average_goals_by_season
  get_season_results(team_id)
  highest_scoring_visitor (bottom 4 lines)
  lowest_scoring_visitor (bottom 4 lines)
  highest_scoring_home (bottom 4 lines)
  lowest_scoring_home (bottom 4 lines)

### CODE REVIEW

File: lib/games_manager.rb [hunk 1]
```diff
@@ -16,9 +16,7 @@ def generate_list(data_path)
   end
 
   def home_and_away_goals_sum
-    games.map do |game|
-      game.away_goals + game.home_goals
-    end
+    games.map {|game| game.away_goals + game.home_goals}
```
Comment (ghost) [inline lib/games_manager.rb]: Very nice

File: lib/games_manager.rb [hunk 2]
```diff
@@ -56,11 +54,8 @@ def percentage_ties
   def count_of_games_by_season
     games_in_season = Hash.new
     games.each do |game|
-      if games_in_season[game.season].nil?
-        games_in_season[game.season] = 1
-      else
-        games_in_season[game.season] += 1
-      end
+      games_in_season[game.season] =0 if games_in_season[game.season].nil?
```
Comment (ghost) [inline lib/games_manager.rb]: Very clean, pop a space between the = and 0

File: lib/games_manager.rb [hunk 3]
```diff
@@ -92,43 +83,20 @@ def get_season_results(team_id)
...
-          result = "W"
-        elsif team_id == game.away_team_id && game.home_goals < game.away_goals
-          result = "W"
-        elsif game.home_goals == game.away_goals
-          result = "T"
-        else
-          result = "L"
-        end
+        result = "non-win"
+        result = "W" if (team_id == game.home_team_id && game.home_goals > game.away_goals) || (team_id == game.away_team_id && game.home_goals < game.away_goals)
```
Comment (ghost) [inline lib/games_manager.rb]: we're going to get dinged on having lines that are too long with this one

File: lib/games_manager.rb [hunk 4]
```diff
@@ -92,43 +83,20 @@ def get_season_results(team_id)
...
-    most = 0
-    best = nil
-    get_season_results(team_id).each do |key, value|   #refactor-able? #I think we can use .valuses.max/.min
-      if value.count('W').to_f / value.size > most
-        most = value.count('W').to_f / value.size
-        best = key
-      end
-    end
-    best
+    (get_season_results(team_id).max_by {|k, v| v.count('W').to_f / v.size})[0]
```
Comment (ghost) [inline lib/games_manager.rb]: I like that it's one line except I have a really hard time reading it.

File: lib/games_manager.rb [hunk 5]
```diff
@@ -171,81 +139,68 @@ def get_season_games(season)
...
+
   def highest_scoring_visitor
     averages = total_away_goals.merge(total_away_games) do |team_id, goals, games|
       (goals/games.to_f).round(2)
     end
-    average_max = averages.max_by do |team_id, average|
-      average
-    end
-    average_max[0]
+    (averages.max_by {|team_id, average| average})[0]
```
Comment (ghost) [inline lib/games_manager.rb]: This refactor and my refactor of the same methods would complement each other nicely!

File: lib/games_manager.rb (hunk 2 above)
Comment (ahenzler) [reply lib/games_manager.rb]: agreed.

File: lib/games_manager.rb (hunk 3 above)
Comment (ahenzler) [reply lib/games_manager.rb]: maybe assign game away/home goals/team id to shorter variables?

File: lib/games_manager.rb [hunk 6]
```diff
@@ -92,43 +83,20 @@ def get_season_results(team_id)
...
         summary[game.season] << result
       end
     end
     summary
   end
 
   def best_season(team_id)
-    most = 0
-    best = nil
-    get_season_results(team_id).each do |key, value|   #refactor-able? #I think we can use .valuses.max/.min
```
Comment (ahenzler) [inline lib/games_manager.rb:113]: values.max seems like a good idea to try here.

File: lib/games_manager.rb [hunk 7]
```diff
@@ -156,11 +124,11 @@ def get_goals_scored(team_id)
   end
 
   def most_goals_scored(team_id)
-    get_goals_scored(team_id).max_by {|score|score.to_i}
+    get_goals_scored(team_id).max_by {|score| score.to_i}
```
Comment (ahenzler) [inline lib/games_manager.rb]: good catch on the spacing here.

**Review by ahenzler — DISMISSED** [review]

**Review by ahenzler — COMMENTED** [review]
Overall really good refactoring done here. There was one method that Aiden also mentioned that was a little hard to follow what was happening in it.

File: lib/games_manager.rb (hunk 4 above)
Comment (denverdevelopments) [reply lib/games_manager.rb]: I share that concern, if it helps we can set it equal to 'best = (get_season_results(team_id).max_by {|k, v| v.count('W').to_f / v.size})[0]'
